### PR TITLE
Update faqs

### PR DESCRIPTION
**Pull request by PBfordev**
I believe that technical documentation should among else have its signal-to-noise ratio as high as possible. This is why I opted to cut out all the information I found not relevant (any more). My changes may seem too radical and I will understand if they are rejected.

But just few examples to (pre)defend myself:
1. Java and .NET are no longer biggest threats to wxWidgets future, hence they do not need to be mentioned.
2. I think that Unicode support is a given so there is no need to answer if it is supported. Similarly, I think it's the best if we forget about DBCS.
3. The "best compiler" section is outdated without mentioning there is a "free" MSVC compiler now as well clang.
4. We do not need to discuss whether it is possible to "compile wxWidgets as a DLL" nor how to reduce executable size.
5. I consider the section about errors with new operator obsolete, as those two defines are now always set to 0.

### Some remaining questions
**General FAQ**
_What is wxWidgets?_ Perhaps mention also at least wxLua and perhaps even wxErlang. I can see that they are both actively maintained; however, I have no knowledge of how usable they are in practice.

_How to use C++ exceptions with wxWidgets?_  Not sure if the text in the second bullet about having to turn exceptions on is still relevant.

**Windows FAQ**
_Why do I get compilation errors when using wxWidgets with DirectShow?_ I have never used DirectX but I suspect this may no longer be relevant and should be removed. However, I am not sure so perhaps somebody else knows?

### CODE REVIEW

**Comment by vadz** [comment]
FWIW I agree with everything you wrote, i.e. I'm perfectly fine with removing all the outdated sections and mentioning wxLua and wxErlang would be nice.

The C++ exceptions part needs to be completely revised: it is now possible to throw from event handlers (and catch these exceptions in `OnExceptionInMainLoop()`) and exceptions are never turned off by default. This might be more suited for the manual than for a FAQ however...

The DirectShow answer looks obsolete to me and IMO should be removed (if nothing else, I've never heard about this problem before, so it looks like it doesn't belong to a _F_AQ list).

Thanks again!

**Comment by PBfordev** [comment]
I have removed that DirectX-related question from the Windows FAQs.

In the General FAQs, I have added mentions of wxLua and wxErlang but removed wxPerl which does not seem to be actively maintained. I have also tried to improve the part discussing exceptions.

File: docs/faq/general/index.md [hunk 1]
```diff
@@ -299,28 +232,27 @@ files such as wx/defs.h, wx/wxchar.h for areas where you'll need to add to
 existing conditionals to set up wide character support and other issues. If the
 GUI runs on a Unix variant, define the `__UNIX__` variable in your makefile.
 
-Then you can start implementing the port, starting with wxWindow,
-wxTopLevelWindow, wxFrame, wxDialog so you can get the minimal sample running
+Then you can start implementing the port, starting with `wxWindow`,
+`wxTopLevelWindow`, `wxFrame`, `wxDialog `so you can get the minimal sample running
```
Comment (PBfordev) [inline docs/faq/general/index.md:236]: Just a reminder to myself
```suggestion
`wxTopLevelWindow`, `wxFrame`, `wxDialog` so you can get the minimal sample running
```

File: docs/faq/windows/index.md [hunk 2]
```diff
@@ -35,12 +27,24 @@ See also [top-level FAQ page](/docs/faq/).
...
-themed automatically so long as you include wx.rc in your own .rc file.
+If the controls do not look as expected (very noticeable e.g. with buttons),
+your application probably does not have an [application manifest](https://docs.microsoft.com/en-us/windows/win32/sbscs/application-manifests)
+and so Windows displays the controls unthemed, as if in Windows 9x.
+wxWidgets provides such a manifest in wx/msw/wx.rc and so your application will be
+themed automatically so long as you include that file in your own .rc file.
+
+<a name="hidpi"></a>
+
+### Why does my application look blurry?
```
Comment (PBfordev) [inline docs/faq/windows/index.md:42]: Just a reminder to myself
```suggestion
### Why does my application look blurry?

```

**Comment by PBfordev** [comment]
I have noticed two tiny formatting issues (which I will fix during the next commit round) but besides that, I think I am done for now.

File: docs/faq/windows/index.md [hunk 3]
```diff
@@ -35,12 +27,24 @@ See also [top-level FAQ page](/docs/faq/).
...
+<a name="manifest"></a>
 
-### What do I need to do for Windows XP?
+### Why do controls in my application look odd?
 
-The XP manifest is included in wx/msw/wx.rc and so your application will be
-themed automatically so long as you include wx.rc in your own .rc file.
+If the controls do not look as expected (very noticeable e.g. with buttons),
+your application probably does not have an [application manifest](https://docs.microsoft.com/en-us/windows/win32/sbscs/application-manifests)
+and so Windows displays the controls unthemed, as if in Windows 9x.
```
Comment (PBfordev) [inline docs/faq/windows/index.md:36]:
```suggestion
and so Windows displays the controls unthemed, as if in Windows 95/98/ME.
```
Change to make this clear even to people who are not familiar with abbreviation "9x".

**Comment by vadz** [comment]
Thanks, I've applied the fixes above and pushed it now.

**Pull request merged**